api/client/ffs.go

```diff
@@ -0,0 +1,244 @@
+package client
```

Comment:
Renamed this file and unfortunately GitHub is showing it as new. Will try to comment on new/important things below.

Reply:
Actually, the API changes will be visible in the `.proto` diff. Just know they're all implemented in here for the client.